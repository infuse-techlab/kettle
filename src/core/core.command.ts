import { Command } from 'commander';
import { Service } from 'typedi';

@Service()
export class CoreCommand extends Command {}
