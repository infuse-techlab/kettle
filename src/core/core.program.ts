import { Service } from 'typedi';
import { name, version } from '../../package.json';
import { ApplicationHandler } from '../application/application.handler';
import { ScriptHandler } from '../application/script/script.handler';
import { CoreCommand } from './core.command';

@Service()
export class CoreProgram {
  private applications: ScriptHandler[] = [];

  constructor(private readonly command: CoreCommand) {
    this.init();
  }

  private init = (): void => {
    this.command.name(name);
    this.command.version(version, '-v, --version', 'output the current version');
    this.command.allowUnknownOption();
    this.command.option('-t, --type <application_type>', 'run a specific application type');
    this.applications = this.loadApplications();
    this.command.parse();
  };

  private loadApplications = (): ScriptHandler[] => {
    return ApplicationHandler.load();
  };

  public run = async (): Promise<void> => {
    const { type } = this.command.opts();
    const applications = this.applications.filter((application) => type === application.getType());
    if (applications.length === 0) {
      return this.command.help();
    }
    return applications[0].run();
  };
}
