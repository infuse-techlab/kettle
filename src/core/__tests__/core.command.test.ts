import { Command } from 'commander';
import { CoreCommand } from '../core.command';

describe('CoreCommand', () => {
  it('should set command version', () => {
    const coreCommand = new CoreCommand();
    expect(coreCommand).toBeInstanceOf(CoreCommand);
    const command = new Command();
    expect(JSON.stringify(coreCommand)).toStrictEqual(JSON.stringify(command));
  });
  it('should have no regression', () => {
    const coreCommand = new CoreCommand();
    expect(coreCommand).toMatchSnapshot();
  });
});
