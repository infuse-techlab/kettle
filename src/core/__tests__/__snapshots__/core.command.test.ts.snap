// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreCommand should have no regression 1`] = `
CoreCommand {
  "_actionHandler": null,
  "_actionResults": Array [],
  "_aliases": Array [],
  "_allowUnknownOption": false,
  "_args": Array [],
  "_combineFlagAndOptionalValue": true,
  "_defaultCommandName": null,
  "_events": Object {},
  "_eventsCount": 0,
  "_executableFile": null,
  "_executableHandler": false,
  "_exitCallback": null,
  "_hasHelpOption": true,
  "_hasImplicitHelpCommand": undefined,
  "_helpCommandDescription": "display help for command",
  "_helpCommandName": "help",
  "_helpCommandnameAndArgs": "help [command]",
  "_helpDescription": "display help for command",
  "_helpFlags": "-h, --help",
  "_helpLongFlag": "--help",
  "_helpShortFlag": "-h",
  "_hidden": false,
  "_maxListeners": undefined,
  "_name": "",
  "_optionValues": Object {},
  "_passCommandToAction": true,
  "_scriptPath": null,
  "_storeOptionsAsProperties": true,
  "_storeOptionsAsPropertiesCalled": false,
  "commands": Array [],
  "options": Array [],
  "parent": null,
  "rawArgs": null,
  Symbol(kCapture): false,
}
`;
